checker/src/org/checkerframework/checker/initialization/InitializationVisitor.java

```diff
@@ -181,6 +185,35 @@ protected boolean checkContract(
                             return true;
                         }
                     }
+                } else {
+                    Set<AnnotationMirror> recvAnnoSet;
+                    @SuppressWarnings("unchecked")
+                    Value value = (Value) store.getValue(fa.getReceiver());
+                    if (value != null) {
+                        recvAnnoSet = value.getAnnotations();
+                    } else if (fa.getReceiver() instanceof LocalVariable) {
+                        Element elem = ((LocalVariable) fa.getReceiver()).getElement();
+                        AnnotatedTypeMirror recvType = atypeFactory.getAnnotatedType(elem);
+                        recvAnnoSet = recvType.getAnnotations();
+                    } else {
+                        // Is there anything better we could do?
```

Comment:
Let's see whether this comes up. It could also be a field access on a local variable or anything more complicated. A general-purpose method for how to use Receiver would be very helpful.